book/src/developing.md

```diff
@@ -0,0 +1,37 @@
+# Developing CSI Driver for Kubernetes
+
+The first step to creating a CSI driver is writing an application implementing the [gRPC](https://grpc.io/docs/) services described in the [CSI specification](https://github.com/container-storage-interface/spec/blob/master/spec.md#rpc-interface)
+
+At a minimum CSI drivers must implement the following CSI services:
+
+* CSI `Identity` service
+  * Enables callers (Kubernetes components and CSI sidecar containers) to identify the driver and what optional functionality it supports.
+* CSI `Node` service
+  * Only `NodePublishVolume`, `NodeUnpublishVolume`, and `NodeGetCapabilities` are required.
+  * Required methods enable callers to make a volume available at a specified path and discover what optional  functionality the driver supports.
+
+All CSI services may be implemented in the same CSI driver application. The CSI driver application should be containerizing to make it easy to deploy on Kubernetes. Once containerized, the CSI driver can be paired with CSI [Sidecar Containers](sidecar-containers.md) and deployed in node or controller mode as appropriate.
```

Comment:
typo: "containerized"

"deployed in node and/or controller"

Reply:
Fixed.